Sources/InputBarItem.swift

```diff
@@ -0,0 +1,260 @@
+/*
+ MIT License
+ 
+ Copyright (c) 2017 MessageKit
+ 
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+ 
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ */
+
+import UIKit
+
+open class InputBarButtonItem: UIButton {
+    
+    public enum Spacing {
+        case fixed(CGFloat)
+        case flexible
+        case none
+    }
+    
+    public typealias InputBarButtonItemAction = ((InputBarButtonItem) -> Void)
+    
+    // MARK: - Properties
+    
+    /// Set automatically when using the InputBarAccessoryView method 'setStackViewItems'
+    open weak var messageInputBar: MessageInputBar?
+    
+    public var spacing: Spacing = .none {
+        didSet {
+            switch spacing {
+            case .flexible:
+                setContentHuggingPriority(1, for: .horizontal)
+            case .fixed:
+                setContentHuggingPriority(1000, for: .horizontal)
+            case .none:
+                setContentHuggingPriority(500, for: .horizontal)
+            }
+        }
+    }
+    
+    /// When not nil this size overrides the intrinsicContentSize
+    open var size: CGSize? = CGSize(width: 20, height: 20) {
+        didSet {
+            invalidateIntrinsicContentSize()
+        }
+    }
+    
+    open override var intrinsicContentSize: CGSize {
+        var contentSize = size ?? super.intrinsicContentSize
+        switch spacing {
+        case .fixed(let width):
+            contentSize.width += width
+        default:
```

Comment:
I don't like using `default` for switching on enum cases. If we add an extra case at a later date we won't get a compile error for it not being handled. No need to change just FYI

Reply:
For sure